sdk/core/System.ClientModel/src/Internal/ContentTypeUtilities.cs

```diff
@@ -0,0 +1,57 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Text;
+
+namespace System.ClientModel.Internal;
+
+/// <summary>
+/// TODO.
+/// </summary>
+internal class ContentTypeUtilities
+{
+    public static bool TryGetTextEncoding(string contentType, out Encoding? encoding)
```

Comment:
TODO - add tests? Move to logging policy?

Reply:
It would be good to look at how widely this is used in Azure.Core to decide about where it should live, and you might be able to move tests in from there as well.

Depending on where it's used in Azure.Core, maybe we can move this type into ClientModel exclusively when we do the Azure.Core 2.0 integration piece?  Or at a minimum, we can track addressing this with https://github.com/Azure/azure-sdk-for-net/issues/41035 and approach holistically later.

Reply:
Looks like it's used in Azure.Core.TestFramework - https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core.TestFramework/src/RecordEntryMessage.cs#L33 and RequestFailedException - https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/src/RequestFailedException.cs#L192